userprofile/urls.py

```diff
@@ -17,5 +17,6 @@
     url(r'^accounts/$', views.ProfileView.as_view(),
         name="profile"),
     url(r'^setlang', views.set_user_language, name="set-user-language"),
+    url(r'^regentoken', views.regenerate_access_token, name="regenerate-access-token"),
```

Comment:
The `setlang` URL has its own history and reasons for its complex implementation. I suspect that `regentoken` does not need such complexity. Refer to the `next` parameter comment in views.py. Basically, I suspect that `regentoken` could include the ending $ in the pattern so that it does not match every URL that starts with `regentoken`.
```suggestion
    url(r'^regentoken/$', views.regenerate_access_token, name="regenerate-access-token"),
```